admin/modules/forum/attachments.php

```diff
@@ -858,14 +858,21 @@ function() {
 			if($num_results > $mybb->input['perpage'])
 			{
 				$pagination_url = "index.php?module=forum-attachments&amp;results=1";
-				$pagination_vars = array('perpage', 'sortby', 'order', 'filename', 'mimetype', 'username', 'fid', 'downloads', 'downloads_dir', 'dateuploaded', 'dateuploaded_dir', 'filesize', 'filesize_dir');
+				$pagination_vars = array('perpage', 'sortby', 'order', 'filename', 'mimetype', 'username', 'downloads', 'downloads_dir', 'dateuploaded', 'dateuploaded_dir', 'filesize', 'filesize_dir');
 				foreach($pagination_vars as $var)
 				{
 					if($mybb->input[$var])
 					{
 						$pagination_url .= "&{$var}=".urlencode($mybb->input[$var]);
 					}
 				}
+				if($mybb->input['forum'])
+				{
+					foreach($mybb->input['forum'] as $fid)
+					{
+						$pagination_url .= "&forum[]=".urlencode($fid);
```

Comment:
Since `fid`s are integers, we really just need to clean them to add them to the URL, nothing should need to be encoded.

```suggestion
						$pagination_url .= "&forum[]=".(int)$fid;
```

Reply:
I think it needs neither urlencode nor a type cast, since if any error exists, we can't get to this point. But it's OK we ensure this variable is an integer.